web/src/markdown.js

```diff
@@ -388,7 +388,26 @@ function make_emoji_span(codepoint, title, alt_text) {
 }
 
 function handleUnicodeEmoji({unicode_emoji, get_emoji_name}) {
-    const codepoint = unicode_emoji.codePointAt(0).toString(16);
+    // We want to avoid turning things like arrows (↔) and keycaps (numbers
+    // in boxes) into qualified emoji (images).
+    // More specifically, we skip anything with text in the second column of
+    // this table https://unicode.org/Public/emoji/1.0/emoji-data.txt
+    // The first regex checks that we have a single unicode character
+    // (which we can't do with `length` because it measures UTF-16 code units).
+    if (/^\P{Emoji_Presentation}\u20E3?$/u.test(unicode_emoji)) {
+        return unicode_emoji;
+    }
+
+    // This unqualifies qualified emoji, which helps us make sure we
+    // can match both versions.
+    // We keep fe0f-20e3 because that pattern shows up in keycaps,
+    // which we leave qualified so that if the user types an unqualified
+    // keycap (e.g. 1 in a box), we don't turn it into an image.
+    unicode_emoji = unicode_emoji.replace(/\uFE0F/, "");
```

Comment:
We do want this unqualified emoji to look it up with `get_emoji_name`. However, if the lookup fails, we need to fall back to returning the original string (qualified or unqualified), so we can’t overwrite it in place.

Reply:
Ah right, thanks